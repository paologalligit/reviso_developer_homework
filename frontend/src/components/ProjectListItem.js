import React from 'react';
import { Button, Grid, List } from 'semantic-ui-react';

import fromMillisToDate from '../utils/convertingTools'

export default ({
  budget,
  name,
  vat,
  penalty,
  date,
  startHour,
  endHour,
}) => (
  <List.Content>
    <List.Header>{`PROJECT: ${name}`}</List.Header>
    <Grid columns={3} divided>
      <Grid.Row>
        <Grid.Column>
          {`Budget: ${budget} €`}
        </Grid.Column>
        <Grid.Column>
          {`Vat: ${vat} €`}
        </Grid.Column>
        <Grid.Column>
          {`Penalty: ${penalty} €`}
        </Grid.Column>
      </Grid.Row>

      <Grid.Row>
        <Grid.Column>
          {`Date: ${fromMillisToDate(date)}`}
        </Grid.Column>
        <Grid.Column>
          {`Start hour: ${startHour}`}
        </Grid.Column>
        <Grid.Column>
          {`End hour: ${endHour}`}
        </Grid.Column>
      </Grid.Row>

      <Grid.Row>
        <Button primary>
          Send Invoice
        </Button>
      </Grid.Row>
    </Grid>
  </List.Content>
);
