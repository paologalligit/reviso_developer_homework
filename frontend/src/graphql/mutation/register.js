import gql from 'graphql-tag';

export default gql`
  mutation(
    $name: String!
    $surname: String!
    $username: String!
    $email: String!
    $password: String!
    $birthdate: String!
    $country: String!
    $city: String!
    $address: String!
    $postal: Int!
  ) {
    registerUser(
      name: $name
      surname: $surname
      username: $username
      email: $email
      password: $password
      birthdate: $birthdate
      country: $country
      city: $city
      address: $address
      postal: $postal
    ) {
      ok
      errors {
        path
        message
      }
    }
  }
`;
