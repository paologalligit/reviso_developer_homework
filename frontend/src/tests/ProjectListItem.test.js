/* eslint-disable func-names */
import 'jsdom-global/register';
import React from 'react';
import toJson from 'enzyme-to-json';
import { mount, render } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';

import ProjectListItem from '../components/ProjectListItem';

describe('project list item suite', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <MockedProvider mocks={[]}>
        <ProjectListItem />
      </MockedProvider>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});

describe('pli interaction', () => {
  it('cannot click button if invoice already sent', () => {
    const wrapper = mount(
      <MockedProvider mocks={[]}>
        <ProjectListItem
          budget={1000}
          name="Test project"
          vat={100}
          penalty={120}
          date="2019-01-01"
          startHour="09:30:00"
          endHour="10:30:00"
          sent
        />
      </MockedProvider>,
    );

    const button = wrapper.find('Button');
    expect(button.instance().props.disabled).toBeTruthy();
  });
});
