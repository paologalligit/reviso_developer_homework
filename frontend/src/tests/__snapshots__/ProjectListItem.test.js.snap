// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project list item suite renders correctly 1`] = `
<div
  class="content"
>
  <div
    class="header"
  >
    PROJECT: undefined
  </div>
  <div
    class="ui divided three column grid"
  >
    <div
      class="row"
    >
      <div
        class="column"
      >
        Budget: undefined €
      </div>
      <div
        class="column"
      >
        Vat: undefined €
      </div>
      <div
        class="column"
      >
        Penalty: undefined €
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="column"
      >
        Date: Invalid Date
      </div>
      <div
        class="column"
      >
        Start hour: undefined
      </div>
      <div
        class="column"
      >
        End hour: undefined
      </div>
    </div>
    <div
      class="row"
    >
      <div>
        <button
          class="ui primary button"
        >
          Send Invoice
        </button>
      </div>
    </div>
  </div>
</div>
`;
